url49526af437_Maria_G_CS325_tgz-pJ8-assignments_farmer_FarmerImplJ8/src/frs/hotgammon/variants/movevalidators/CompleteMoveValidator.java

```diff
@@ -0,0 +1,254 @@
+package frs.hotgammon.variants.movevalidators;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import frs.hotgammon.MoveValidator;
+import frs.hotgammon.framework.Color;
+import frs.hotgammon.framework.Game;
+import frs.hotgammon.framework.GameObserver;
+import frs.hotgammon.framework.Location;
+
+public class CompleteMoveValidator implements MoveValidator {
+
+	private Game game;
+	
+	public CompleteMoveValidator() {
+	}
+	
+	public CompleteMoveValidator(Game game) {
+		this.setGame(game);
+	}
+	
+	@Override
+	public boolean isValid(Location from, Location to) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 68-80, lines 81-134 into separate methods.